Sources/NIOHTTP1/HTTPUpgradeHandler.swift

```diff
@@ -236,8 +236,8 @@ public class HTTPServerUpgradeHandler: ChannelInboundHandler {
                     if bufferedMessages.count > 0 {
                         ctx.fireChannelReadComplete()
                     }
-                }.then {
-                    ctx.pipeline.remove(ctx: ctx)
+                }.whenComplete {
```

Comment:
This is not quite semantically equivalent, but I think it's more accurate anyway.